native_libs/src/2d_array.c

```diff
@@ -72,7 +73,7 @@ char **drop_row(char **mat, size_t rows, size_t cols, size_t stride_r, size_t st
         for(int i = 0; i < row_to_drop; i++) {
                 rows_to_copy[i] = i;
         }
-        for(int i = row_to_drop+1; i <= rows; i++) {
```

Comment:
This is not related to this change, but to dataframes in general: I'm not really sure this copying of rows is good/necessary. I would create an indexing vector/array containing the mapping between row indices and the real indices in the array of arrays.
So let's say I have an array:
```
df = [[11, 12, 13], [21, 22, 23], [31, 32, 33]]
```
I suggest we create a 1D vector `rowIndex = [0, 1, 2]` and calculate: `df[i] ==> df[rowIndex[i]]`.
No when we want to remove a row, say the `[21, 22, 23]` one, it suffices to set `rowIndex = [0, 2, -1]` so that `df[1]` now translates to `df[rowIndex[1] ==> 2] ==> df[2]`. This also allows you to, in an optimistic case, remove+add a row with zero allocations ;)

Rationale: as a user, I expect the deletion of a row to be an `O(1)` kind of operation. Now removing the first row does a lot of copying/allocating.

This doesn't need to be done in this PR or even this year, but keep that in mind as an idea ;)

Reply:
Oh btw, you could do the same for columns.

BTW2 pandas sure keeps an indexing vector.

Reply:
Oh and I can see that we're trying to be Haskellish here and don't change the dataframes in place. I'm just not sure that's really possible for really large datasets.

Reply:
Ok, at this point we need to deliver a basic library where we can show how to work with data in Luna. It is more about showing possibilities than working with big data. Because we don't know how the use cases will look like I would suggest to leave it for now and work on it in the next iteration.

Reply:
@mwu-tow @piotrMocz

Reply:
About Array2D it is a general usage implementation for flat data with strides and offsets - to have a fast access to the certain views.